src/lib.rs

```diff
@@ -2208,14 +2208,13 @@ pub trait Parser<'a, I: Input<'a>, O, E: ParserExtra<'a, I> = extra::Default>:
     /// );
     /// ```
     #[cfg(feature = "pratt")]
-    fn pratt<InfixOps, InfixOpsOut>(
+    fn pratt<InfixParser, InfixOperator>(
         self,
-        ops: InfixOps,
-    ) -> Pratt<I, O, E, Self, pratt::Infix<InfixOps, InfixOpsOut>>
+        ops: InfixParser,
+    ) -> Pratt<I, O, E, Self, pratt::Infix<InfixParser, InfixOperator>>
     where
         I: Input<'a>,
         E: ParserExtra<'a, I>,
-        InfixOps: Parser<'a, I, InfixOpsOut, E>,
```

Comment:
Any particular reason this trait bound has to disappear? It was originally added with the idea that it makes no sense to construct a `Parser` which cannot parse.

Reply:
It doesn't have to disappear, but it does have to change a bit. At first I didn't figure out how to change it, so I haphazardly deleted it.